search/src/Components/upload/uploader_category_mapper.js

```diff
@@ -52,14 +52,14 @@ class CategoryMapper extends React.Component {
 
     changeRole(e, index) {
         this.modifyCategories(index, "role", e.target.value);
-        this.props.handleCategoriesSaved(false);
-        this.props.handleErrorMessage("init");
+        this.props.handleValidation('categories', false);
```

Comment:
Personally, I think passing back the step name from the component is a flaw in separation of concerns. The component implementing the step should not need to know its step name for the sake of the container (upload stepper's) record keeping.